### PR TITLE
Docker event connection test

### PR DESCRIPTION
**Pull request by eadventurous**
Signed-off-by: Egor Ivkov <e.o.ivkov@gmail.com>

### CODE REVIEW

File: iroha_network/src/lib.rs [hunk 1]
```diff
@@ -165,31 +166,33 @@ impl Connection {
     }
 }
 
-impl Stream for Connection {
+impl Iterator for Connection {
     type Item = Vec<u8>;
 
-    fn poll_next(mut self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+    fn next(&mut self) -> Option<Self::Item> {
```
Comment (humb1t) [inline iroha_network/src/lib.rs:172]: Wouldn't it be the problem when we will replace std::net::TcpStream with async_std::net::TcpStream?

Comment (eadventurous) [reply iroha_network/src/lib.rs:172]: Then we should change it back to stream when we introduce that change. Here there was a problem with the fact that we wrapped synchronous blocking code with async stream. And it just did not execute, I think the async_std scheduler was blocked by this call and could not schedule other tasks properly.

Comment (humb1t) [reply iroha_network/src/lib.rs:172]: ok, ty

File: iroha_client/tests/subscribe_to_changes.rs [hunk 2]
```diff
@@ -104,24 +113,33 @@ mod tests {
...
-                        println!("Entity changed: {:?}", entity);
-                        return ();
+        task::spawn(async move {
+            task::sleep(Duration::from_millis(300)).await;
+            iroha_client
+                .submit(create_asset.into())
+                .await
+                .expect("Failed to prepare state.");
+        });
+        async_std::future::timeout(
```
Comment (humb1t) [inline iroha_client/tests/subscribe_to_changes.rs:123]: For such cases you should import `use async_std::feature` and invoke it via `future::timeout`

File: iroha_client_cli/src/main.rs [hunk 3]
```diff
@@ -443,12 +443,12 @@ mod maintenance {
                 #[allow(clippy::empty_loop)]
                 loop {}
             });
-            task::sleep(Duration::from_millis(300)).await;
+            std::thread::sleep(Duration::from_millis(300));
```
Comment (humb1t) [inline iroha_client_cli/src/main.rs:446]: `thread::sleep`

**Pull request merged**